test/e2e/proxy_test.go

```diff
@@ -61,10 +65,126 @@ func proxyTestSuite(t *testing.T) {
 	// the deletion of TrustedCAConfigMap, which the former relies on
 	t.Run("Certificate validation", tc.testCertsImport)
 	t.Run("Environment variables validation", tc.testEnvVars)
+
+	t.Run("External requests respect global proxy", tc.testProxyRequests)
+
 	t.Run("Certificate validation removal", tc.testCertsRemoval)
 	t.Run("Environment variables removal validation", tc.testEnvVarRemoval)
 }
 
+// testProxyRequest tests that external requests from each Windows node properly go through the proxy
+// and circumvent the proxy if the target address is specified in the NO_PROXY list
+func (tc *testContext) testProxyRequests(t *testing.T) {
+	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
+	require.NoError(t, err)
+
+	// variables to look for in the curl output
+	httpVal := fmt.Sprintf("http_proxy == '%s'", clusterProxy.Status.HTTPProxy)
+	noProxyVal := fmt.Sprintf("no_proxy == '%s'", clusterProxy.Status.NoProxy)
```

Comment:
please pass in the URLs, and do this string creation within verifyCurlOutput. No need for this to live outside of it.